crypto/src/merkle/mod.rs

```diff
@@ -127,6 +127,39 @@ impl<H: Hasher> MerkleTree<H> {
         Ok(MerkleTree { nodes, leaves })
     }
 
+    /// Returns new Merkle tree built from default, empty leaves using hash function
+    /// specified by the `H` generic parameter.
+    pub fn build_empty(depth: usize) -> Result<Self, MerkleTreeError> {
```

Comment:
The function is actually never erroring, so we can return `Self` directly instead of `Result`.